virttest/test_setup.py

```diff
@@ -1438,6 +1438,17 @@ def check_vfs_count(self):
         expected_count = int((re.findall("(\d+)", self.driver_option)[0])) * len(self.get_pf_ids())
         return (self.get_vfs_count() == expected_count)
 
+    def get_controller_type(self):
+        """
+        Get the Controller Type for SR-IOV Mellanox Adapter PFs
+        """
+        try:
+            cmd = "lspci | awk '/%s/ {print $2}'" % self.pf_filter_re
+            return process.system_output(cmd, shell=True).strip().split("\n")[-1]
```

Comment:
This would make the API to retrieve better output

```python
cmd = "lspci | grep '%s'| grep -o '\s[A-Z].*:\s'" % self.pf_filter_re
return process.system_output(cmd, shell=True).split("\n")[-1].strip().strip(':')
```